src/injector/src/main.rs

```diff
@@ -105,43 +105,111 @@ fn start_seed_registry() {
                     .with_additional_header("X-Content-Type-Options", "nosniff")
                 },
 
-                (GET) (/v2/registry/manifests/{_tag :String}) => {
-                    handle_get_manifest(&root)
+                (GET) (/v2/{name :String}/manifests/{tag :String}) => {
+                    handle_get_manifest(&root, &name, &tag)
                 },
 
-                (GET) (/v2/{_namespace :String}/registry/manifests/{_ref :String}) => {
-                    handle_get_manifest(&root)
+                (GET) (/v2/{namespace :String}/{name :String}/manifests/{tag :String}) => {
+                    let namespaced_name :String = namespace + "/" + &name;
+                    handle_get_manifest(&root, &namespaced_name, &tag)
                 },
 
-                (HEAD) (/v2/registry/manifests/{_ref :String}) => {
+                (GET) (/v2/{n1 :String}/{n2 :String}/{name :String}/manifests/{tag :String}) => {
+                    let namespaced_name :String = n1 + "/" + &n2 + "/" + &name;
+                    handle_get_manifest(&root, &namespaced_name, &tag)
+                },
+
+                (GET) (/v2/{n1 :String}/{n2 :String}/{n3 :String}/{name :String}/manifests/{tag :String}) => {
+                    let namespaced_name :String = n1 + "/" + &n2 + "/" + &n3 + "/" + &name;
+                    handle_get_manifest(&root, &namespaced_name, &tag)
+                },
+
+                (GET) (/v2/{n1 :String}/{n2 :String}/{n3 :String}/{n4 :String}/{name :String}/manifests/{tag :String}) => {
+                    let namespaced_name :String = n1 + "/" + &n2 + "/" + &n3 + "/" + &n4 + "/" + &name;
+                    handle_get_manifest(&root, &namespaced_name, &tag)
+                },
```

Comment:
I hate this, but it will work for most namespaced containers.

Reply:
A potentially better pattern would be to use the `!find_route` macro https://docs.rs/rouille/latest/rouille/macro.find_route.html and create our own custom router so that we could handle arbitrary path segments / wildcards on our own.

Reply:
Lol I hate it too - I'll give find_route a try though

Reply:
Resolved